docs/man/rpmbuild.1.scd

```diff
@@ -190,6 +185,64 @@ _TAR_ARCHIVE_
 
 See *rpm-common*(8) for the options common to all operations.
 
+# BUILD PROCESS
+
+Packages are built in a number of logical steps, outlined below. The steps that
+produce build artifacts (including the final packages) or diagnostic output are
+called _stages_, each denoted by a single-letter identifier in parentheses. You
+can instruct *rpmbuild* to stop after completing a specific stage by passing it
+the respective identifier (see also *OPERATIONS*).
+
+Some steps are implemented as standalone scripts whose contents is dumped in the
+*rpmbuild* output for diagnostic purposes. Their names are shown in parentheses
+after the stage identifier (if any), prefixed with *%* if they correspond to a
+spec section or directive.
+
+Binary packages are produced separately for one or more platforms, called _build
+targets_. By default, the current platform is the only build target. This can be
+overridden with *--target* or by declaring a *BuildArch* in the spec. Currently,
+using a *--target* and *BuildArch* that disagree with each other will result in
+undefined behavior.
+
+Note that *BuildArch* causes the spec parsing to recurse from the start, causing
+any macros before that line to be expanded twice. This can yield unexpected
+results, in particular with *%global*.
```

Comment:
Perhaps this should go into a BUGS section to make it easier to discover? It's a common gotcha for packagers and something we wont be able to cure.